### PR TITLE
Fixes #281: Add Aura integration tests

### LINKED ISSUE #281: Aura Integration Tests in CI Chain

### PR DESCRIPTION
**Pull request by utnaf**
Fixes #281 

Add Aura integration tests.

### CODE REVIEW

**Comment by utnaf** [comment]
Regarding #289 I successfully executed some local tests against the Neo4j 4.3 instance that @moxious provided

fyi @conker84

File: spark-2.4/src/test/scala/org/neo4j/spark/SparkConnectorAuraTest.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import org.neo4j.driver._
+import org.neo4j.spark.SparkConnectorAuraTest._
+
+object SparkConnectorAuraTest {
+  private var neo4j: Driver = _
+  private val username: Option[String] = Option[String](System.getenv("AURA_USER"))
+  private val password: Option[String] = Option[String](System.getenv("AURA_PASSWORD"))
+  private val url: Option[String] = Option[String](System.getenv("AURA_URI"))
+
+  val sparkSession = SparkSession.builder()
```
Comment (conker84) [inline spark-2.4/src/test/scala/org/neo4j/spark/SparkConnectorAuraTest.scala]: please create the `sparkSession` after the assume checks in the `BeforeClass` and tear it down in the `AfterClass`

File: spark-2.4/src/test/scala/org/neo4j/spark/SparkConnectorAuraTest.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    assumeTrue(username.isDefined)
+    assumeTrue(password.isDefined)
+    assumeTrue(url.isDefined)
+
+    neo4j = GraphDatabase.driver(url.get, AuthTokens.basic(username.get, password.get))
+  }
+
+  @AfterClass
+  def tearDown(): Unit = {
+    if(neo4j.isInstanceOf[Driver]) {
```
Comment (conker84) [inline spark-2.4/src/test/scala/org/neo4j/spark/SparkConnectorAuraTest.scala:34]: do we need a null check?

File: spark-2.4/src/test/scala/org/neo4j/spark/SparkConnectorAuraTest.scala (hunk 1 above)
Comment (conker84) [reply spark-2.4/src/test/scala/org/neo4j/spark/SparkConnectorAuraTest.scala]: also in spark 3.0

File: spark-2.4/src/test/java/org/neo4j/spark/DataSourceReaderTypesTSE.java [hunk 3]
```diff
@@ -289,6 +289,7 @@ public void testReadNodeWithMap() {
         SparkConnectorScalaSuiteIT.session()
                 .writeTransaction(transaction -> transaction.run(query).consume());
 
+
```
Comment (conker84) [inline spark-2.4/src/test/java/org/neo4j/spark/DataSourceReaderTypesTSE.java]: can you revert it?

**Review by conker84 — CHANGES_REQUESTED** [review]
@utnaf please look at my comments

**Pull request merged**